interfaces/builtin/basedeclaration.go

```diff
@@ -326,6 +326,10 @@ slots:
       slot-snap-type:
         - core
     deny-auto-connection: true
+  media-hub:
+    allow-installation:
+      slot-snap-type:
+        - app
```

Comment:
Thanks for removing the deny-auto-connection. I forgot to mention that since this is not an implicit slot, we need a deny-connection constraint. It also occurred to me that you are supporting talking to the deb on classic so this needs to be adjusted to be:
```
  media-hub:
    allow-installation:
      slot-snap-type:
        - app
        - core
    deny-connection:
      on-classic: false
```

This is saying that on core, allow connections to the snap and on all-snaps a snap declaration is required to connect to the snap. When connection is allowed, allow auto-connection.